libraries/SD/src/utility/SdFat.h

```diff
@@ -431,14 +431,16 @@ union cache_t {
 class SdVolume {
  public:
   /** Create an instance of SdVolume */
-  SdVolume(void) :allocSearchStart_(2), fatType_(0) {}
+  SdVolume(void) :allocSearchStart_(2), fatType_(0) { if (!cacheBuffer_) cacheBuffer_ = (cache_t *)malloc(sizeof(cache_t)); }
+  /** Delete an instance of SdVolume */
+  ~SdVolume() { free(cacheBuffer_); cacheBuffer_ = NULL; }

```

Comment:
The original cache was a static array, accessible from any instance. This change makes it a static pointer, and the constructor checks if it has been allocated before, in which case nothing is done, or not, in which case it is allocated.
This destructor just blindly frees the mem. If there is more than one instance, the others will then try to access after free.
I think it's unlikely that there will be more than one instance of this class, but I think this is something that should be fixed to cover e.g.: temps (pass by value) and copy. 
I suggest use of a shared_ptr.

Reply:
I had only considered the single global SD object.  Since then, with the BSSL stuff I've been introduced to the wonders of the std::shared_ptr and this would be a good use of one like you suggest.  Will see if I can fix it up this weekend.

Reply:
Done, but using a separate refcnt instead of std::ptr as there needs to be a special-case for cnt==1 (like the BearSSL issue).